addons/im_livechat/static/src/models/public_livechat_global_notification_handler.js

```diff
@@ -0,0 +1,102 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { one } from '@mail/model/model_field';
+
+import session from 'web.session';
+import utils from 'web.utils';
+
+registerModel({
+    name: 'PublicLivechatGlobalNotificationHandler',
```

Comment:
Is there a reason to create a new model rather than extending `MessagingNotificationHandler`?

Reply:
It's unsafe for the moment to reuse existing models. I make a dedicated model just so that everything about public livechat is in their own models. Eventually will be integrated with `MessagingNotificationHandler`, but I prefer to play safe and do it step by step.

The routes and logic are not the same, so it's less effort to keep doing structural changes for the moment.